lib/node/runner/node_runner.go

```diff
@@ -651,11 +663,35 @@ func (nr *NodeRunner) proposeNewBallot(round uint64) (ballot.Ballot, error) {
 
 	nr.log.Debug("new ballot created", "ballot", theBallot)
 
-	nr.ConnectionManager().Broadcast(*theBallot)
+	//nr.ConnectionManager().Broadcast(*theBallot)
+	nr.broadcastBallot(*theBallot)
 
 	return *theBallot, nil
 }
 
 func (nr *NodeRunner) NodeInfo() node.NodeInfo {
 	return nr.nodeInfo
 }
+
+func (nr *NodeRunner) startBroadcastBallot() {
+	for {
+		select {
+		case b := <-nr.ballotBroadcastChannel:
+			nr.broadcastBallot(b)
+		}
+	}
+}
+
+func (nr *NodeRunner) BroadcastBallot(b ballot.Ballot) {
+	go func() {
+		nr.ballotBroadcastChannel <- b
+	}()
+}
+
+func (nr *NodeRunner) broadcastBallot(b ballot.Ballot) {
+	go func() {
+		nr.handleBallotMessage(b)
```

Comment:
I have a question:  Why node do `broadcastBallot` and `handleBallotMessage` at the same time? :->  It means that `b ballot.Ballot` is not from node itself?